### ISSUE TITLE
Causes kotlin to be unusable in IDEA 2020.2.1

### ISSUE THREAD

**Issue by trevjonez** [detekt/detekt-intellij-plugin#100]
Just going to cross post the issue that got me thinking about trying to disable plugins to get my IDE to behave correctly.

https://youtrack.jetbrains.com/issue/IDEA-249231

my guess is you need to up the kotlin version to 1.4

**Comment by arturbosch** [comment]
Thanks for reporting this.
I can verify that something beaks in the plugin with Kotlin 1.4.0:

```
Expecting:
  <java.lang.NoSuchMethodError: 'void kotlin.jvm.internal.FunctionReferenceImpl.<init>(int, java.lang.Class, java.lang.String, java.lang.String, int)'>
to be an instance of:
  <java.lang.ClassCastException>
but was:
  <"java.lang.NoSuchMethodError: 'void kotlin.jvm.internal.FunctionReferenceImpl.<init>(int, java.lang.Class, java.lang.String, java.lang.String, int)'
	at io.gitlab.arturbosch.detekt.api.internal.CommaSeparatedPattern$mapToRegex$1.<init>(CommaSeparatedPattern.kt)
	at io.gitlab.arturbosch.detekt.api.internal.CommaSeparatedPattern$mapToRegex$1.<clinit>(CommaSeparatedPattern.kt)
	at io.gitlab.arturbosch.detekt.api.internal.CommaSeparatedPattern.mapToRegex(CommaSeparatedPattern.kt:10)
	at io.gitlab.arturbosch.detekt.core.config.DefaultPropertiesConfigValidator$validate$1.invoke(DefaultPropertiesConfigValidator.kt:24)
```

**Comment by arturbosch** [comment]
FIxed in detekt 1.14.1 works with included build.

**Comment by jonasfa** [comment]
For those who are investigating the same exception, the change that fixed this issue replaced a constructor reference with a lambda: https://github.com/detekt/detekt/pull/3107/files
